dds/idl/typeobject_generator.cpp

```diff
@@ -519,18 +519,245 @@ operator<<(std::ostream& out, const OpenDDS::XTypes::TypeIdentifier& ti)
   return out;
 }
 
+namespace {
+template<typename T>
+std::string getTypeName();
+
+template<>
+std::string getTypeName<OpenDDS::DCPS::String>()
+{
+  return "String";
+}
+
+template<>
+std::string getTypeName<OpenDDS::XTypes::AppliedAnnotationParameterSeq>()
+{
+  return "XTypes::AppliedAnnotationParameterSeq";
+}
+
+template<>
+std::string getTypeName<OpenDDS::XTypes::AnnotationParameterValue>()
+{
+  return "XTypes::AnnotationParameterValue";
+}
+
+template<>
+std::string getTypeName<OpenDDS::XTypes::AppliedBuiltinMemberAnnotations>()
+{
+  return "XTypes::AppliedBuiltinMemberAnnotations";
+}
+
+template<>
+std::string getTypeName<OpenDDS::XTypes::AppliedAnnotationSeq>()
+{
+  return "XTypes::AppliedAnnotationSeq";
+}
+
+template<>
+std::string getTypeName<OpenDDS::XTypes::AppliedVerbatimAnnotation>()
+{
+  return "XTypes::AppliedVerbatimAnnotation";
+}
+
+template<>
+std::string getTypeName<OpenDDS::XTypes::AppliedBuiltinTypeAnnotations>()
+{
+  return "XTypes::AppliedBuiltinTypeAnnotations";
+}
+
+template<>
+std::string getTypeName<OpenDDS::XTypes::CompleteTypeDetail>()
+{
+  return "XTypes::CompleteTypeDetail";
+}
+}
+
+template<typename T>
+std::ostream&
+operator<<(std::ostream& out, const OpenDDS::XTypes::Optional<T>& optional)
+{
+  if (optional.present) {
+    return out
+      << "XTypes::Optional<" << getTypeName<T>()  << ">("
+      << optional.value
+      << ")";
+  } else {
+    return out << "XTypes::Optional<" << getTypeName<T>() << ">()";
+  }
+}
+
+std::ostream&
+operator<<(std::ostream& out, const OpenDDS::XTypes::AppliedVerbatimAnnotation& ann)
+{
+  return out
+    << "XTypes::AppliedVerbatimAnnotation("
+    << ann.placement << ", "
+    << ann.language << ", "
+    << ann.text
+    << ")";
+}
+
+std::ostream&
+operator<<(std::ostream& out, const OpenDDS::XTypes::AppliedBuiltinTypeAnnotations& ann)
+{
+  return out
+    << "XTypes::AppliedBuiltinTypeAnnotations("
+    << ann.verbatim
+    << ")";
+}
+
+std::ostream&
+operator<<(std::ostream& out, const OpenDDS::XTypes::NameHash& name_hash)
+{
+  for (size_t i = 0; i < sizeof name_hash; ++i) {
+    out << int(name_hash[i]);
+    if (i < sizeof name_hash - 1) {
+      out << ", ";
+    }
+  }
+  return out;
+}
+
+std::ostream&
+operator<<(std::ostream& out, const OpenDDS::XTypes::AnnotationParameterValue& param_value)
+{
+  out << "XTypes::AnnotationParameterValue(";
+  switch (param_value.kind) {
+  case OpenDDS::XTypes::TK_BOOLEAN:
+    out << (param_value.boolean_value ? "true" : "false");
+    break;
+  case OpenDDS::XTypes::TK_BYTE:
+    out << "static_cast<ACE_CDR::Octet>(" << param_value.byte_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_INT16:
+    out << "static_cast<ACE_CDR::Short>(" << param_value.int16_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_UINT16:
+    out << "static_cast<ACE_CDR::UShort>(" << param_value.uint16_value << ")";
+    break;
+  case OpenDDS::XTypes::TK_INT32:
+    out << "XTypes::TK_INT32, " << param_value.int32_value;
```

Comment:
This looks wrong.  Use same form as others.

Reply:
Good catch. Thanks.